### PR TITLE
ci: added security vulnerability scanning using trivy and updated rel…

### PR DESCRIPTION
**Pull request by niravparikh05**
### What does this PR change?

- added security vulnerability scanning using trivy and updated release workflow

### Does the PR depend on any other PRs or Issues? If yes, please list them.

- NA

### Checklist

I confirm, that I have...

- [ ] Read and followed the contributing guide in `CONTRIBUTING.md`
- [ ] Added tests for this PR
- [ ] Formatted the code using `go fmt` (if applicable)
- [ ] Updated [documentation on the Paralus docs site](https://github.com/paralus/website/blob/main/docs) (if applicable)
- [ ] Updated `CHANGELOG.md`

### CODE REVIEW

File: .github/workflows/security.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        uses: aquasecurity/trivy-action@0.10.0
+        with:
+          image-ref: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{ github.sha }}
+          format: 'table'
+          exit-code: '1'
+          ignore-unfixed: true
+          vuln-type: 'os,library'
+          severity: 'CRITICAL,HIGH'
+
+      - name: Build an image from Dockerfile (paralus-init)
```
Comment (akshay196) [inline .github/workflows/security.yml:59]: The github action runtime can be optimized by running parlaus-init and kratos-sync as separate jobs.

File: .github/workflows/security.yml [hunk 2]
```diff
@@ -0,0 +1,89 @@
+name: Scan container images
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  schedule:
+    - cron: '0 0 * * 1'
```
Comment (akshay196) [inline .github/workflows/security.yml:9]: This will run at midnight UTC (5:30 IST). How about running scan at 9:30 IST so it would be convenient for us?

**Review by akshay196 — CHANGES_REQUESTED** [review]
Attaching security.yml with requested changes here.
[security.yml.txt](https://github.com/paralus/paralus/files/11557402/security.yml.txt)

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -30,7 +29,7 @@ jobs:
       - name: Set up Go
         uses: actions/setup-go@v4
         with:
-          go-version: ${{ matrix.go-version }}
+          go-version-file: 'go.mod'
 
       # Sanity check before publishing
```
[the remark this answers is not in the document]
Comment (niravparikh05) [reply .github/workflows/release.yml:34]: removed this as it is already being performed in go.yml

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -40,15 +39,15 @@ jobs:
         run: go test -v ./...
 
       - name: Log in to the Container registry
-        uses: docker/login-action@f4ef78c080cd8ba55a85445d5b36e214a81df20a
+        uses: docker/login-action@v2
```
[the remark this answers is not in the document]
Comment (niravparikh05) [reply .github/workflows/release.yml:42]: login-action is not required here, removed it.

File: .github/workflows/release.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (niravparikh05) [reply .github/workflows/release.yml]: moved it above test all step

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -40,15 +39,15 @@ jobs:
...
         with:
           registry: ${{ env.REGISTRY }}
           username: ${{ secrets.REGISTRY_USERNAME }}
           password: ${{ secrets.REGISTRY_PASSWORD }}
 
       - name: Extract metadata (tags, labels) for Docker
         id: meta
-        uses: docker/metadata-action@c4ee3adeed93b1fa6a762f209fb01608c1a22f1e
+        uses: docker/metadata-action@v4
```
[the remark this answers is not in the document]
Comment (niravparikh05) [reply .github/workflows/release.yml]: pinned it to sha as of now however docker verified github actions should be fine when used with versions I suppose.

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -68,7 +67,7 @@ jobs:
 
       - name: Extract metadata (tags, labels) for Docker (paralus-init)
         id: meta-init
-        uses: docker/metadata-action@c4ee3adeed93b1fa6a762f209fb01608c1a22f1e
+        uses: docker/metadata-action@v4
```
[the remark this answers is not in the document]
Comment (niravparikh05) [reply .github/workflows/release.yml]: pinned it to sha as of now however docker verified github actions should be fine when used with versions I suppose.

File: .github/workflows/release.yml [hunk 7]
```diff
@@ -99,11 +98,11 @@ jobs:
             type=sha
 
       - name: Build and push Docker image (kratos-sync)
-        uses: docker/build-push-action@3b5e8027fcad23fda98b2e3ac259d8d67585f671
+        uses: docker/build-push-action@v4
```
[the remark this answers is not in the document]
Comment (niravparikh05) [reply .github/workflows/release.yml]: pinned it to sha as of now however docker verified github actions should be fine when used with versions I suppose.

File: .github/workflows/security.yml [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version-file: 'go.mod'
+
+      # Sanity check before publishing
+      - name: Create k8s Kind Cluster
+        uses: helm/kind-action@v1.2.0
+
+      - name: Test all
```
[the remark this answers is not in the document]
Comment (niravparikh05) [reply .github/workflows/security.yml]: Agreed, not needed here.

File: .github/workflows/security.yml (hunk 2 above)
Comment (niravparikh05) [reply .github/workflows/security.yml]: this works too

File: .github/workflows/release.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (niravparikh05) [reply .github/workflows/release.yml]: pinned it to sha as of now however docker verified github actions should be fine when used with versions I suppose.

File: .github/workflows/security.yml [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Set up Go
```
Comment (akshay196) [inline .github/workflows/security.yml:27]: This step can be removed because we no longer running go test.

File: .github/workflows/security.yml [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+        uses: actions/setup-go@v4
+        with:
+          go-version-file: 'go.mod'
+
+      - name: Build an image from Dockerfile
+        run: |
+          docker build -f "Dockerfile" -t ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${{ github.sha }} .
```
Comment (akshay196) [inline .github/workflows/security.yml:34]: We can just tag image as `paralus:${{ github.sha }}` as we are not pushing image to registry. With this, we no longer need to set `REGISTRY` and `IMAGE_NAME` env variables above - lesser lines.

**Review by akshay196 — CHANGES_REQUESTED** [review]

File: .github/workflows/security.yml (hunk 9 above)
Comment (nirav-rafay) [reply .github/workflows/security.yml]: removed

File: .github/workflows/security.yml (hunk 10 above)
Comment (nirav-rafay) [reply .github/workflows/security.yml]: renamed the tag

**Review by akshay196 — APPROVED** [review]

**Pull request merged**